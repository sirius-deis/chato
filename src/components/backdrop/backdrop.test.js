import { render, screen, fireEvent } from "@testing-library/react";
import Backdrop from "./backdrop";

describe("Backdrop component", () => {
  it("should match snapshot", () => {
    const { container } = render(<Backdrop></Backdrop>);
    expect(container).toMatchSnapshot();
  });
  it("should have a dark background", () => {
    render(<Backdrop></Backdrop>);
    expect(screen.getByTestId("backdrop")).toHaveStyle({ backgroundColor: "var(--shadow-color)" });
  });
  it("should be clickable", () => {
    const handleClick = jest.fn();
    render(<Backdrop onClickHandler={handleClick}></Backdrop>);
    fireEvent.click(screen.getByTestId("backdrop"));
    expect(handleClick).toHaveBeenCalled();
  });
})