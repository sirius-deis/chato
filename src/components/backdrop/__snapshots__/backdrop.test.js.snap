// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop component should match snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg eORFT"
    data-testid="backdrop"
  />
</div>
`;
